raspberry-pi/4/i2c.nix

```diff
@@ -21,26 +22,52 @@ let
 in
 {
   options.hardware.raspberry-pi."4" = {
-    i2c0.enable = lib.mkEnableOption ''
-      Turn on the VideoCore I2C bus (maps to /dev/i2c-22) and enable access from the i2c group.
-      After a reboot, i2c-tools (e.g. i2cdetect -F 22) should work for root or any user in i2c.
-    '';
-    i2c1.enable = lib.mkEnableOption ''
-      Turn on the ARM I2C bus (/dev/i2c-1 on GPIO pins 3 and 5) and enable access from the i2c group.
-      After a reboot, i2c-tools (e.g. i2cdetect -F 1) should work for root or any user in i2c.
-    '';
+    i2c0 = {
+      enable = lib.mkEnableOption ''
+        Turn on the VideoCore I2C bus (maps to /dev/i2c-22) and enable access from the i2c group.
+        After a reboot, i2c-tools (e.g. i2cdetect -F 22) should work for root or any user in i2c.
+      '';
+      frequency = lib.mkOption {
+        type = lib.types.int;
+        default = 100000;
```

Comment:
I assume this is the default frequency for raspberry pi just now?

Reply:
https://github.com/raspberrypi/linux/blob/rpi-5.15.y/arch/arm/boot/dts/bcm2711-rpi-4-b.dts#L490-L498

Yes, but I think I like @06kellyjac proposal more, where we wouldn't need to be making assumptions like that.